docs/resources/aws_emr_cluster.md

```diff
@@ -0,0 +1,106 @@
+|kerberos_attributes_realm               | The name of the Kerberos realm to which all nodes in a cluster belong.|
+|kerberos_attributes_realm_ad_domain_join_user  | A user with sufficient privileges to join resources to the domain. Required only when establishing a cross-realm trust with an Active Directory domain.|
+|log_encryption_kms_key_id               | The KMS key used for encrypting log files.|
+|log_uri                                 | The path to the Amazon S3 location where logs for this cluster are stored.|
+|release_label                           | The Amazon EMR release label, which determines the version of open-source application packages installed on the cluster.|
+|scale_down_behavior                     | The way that individual Amazon EC2 instances terminate when an automatic scale-in activity occurs or an instance group is resized. Valid values are TERMINATE_AT_INSTANCE_HOUR, TERMINATE_AT_TASK_COMPLETION |
+|service_role                            | The IAM role that Amazon EMR assumes in order to access Amazon Web Services resources on your behalf.|
+|step_concurrency_level                  | Specifies the number of steps that can be executed concurrently.|
+|visible_to_all_users                    | Indicates whether the cluster is visible to IAM principals in the Amazon Web Services account associated with the cluster.|
+|managed_scaling_policy_unit_type        | The unit type used for specifying a managed scaling policy. Valid values are InstanceFleetUnits, Instances, VCPU|
+              
+## Examples
+
+
+##### Test that an EMR Cluster does not exist
+```ruby
+describe aws_emr_cluster(cluster_id: 'INVALID_CLUSTER_ID') do
+  it { should_not exist }
+end
+```
+##### Test that an EMR Cluster state is `WAITING`
+```ruby
+describe aws_emr_cluster('CLUSTER_ID') do
+  its ('state') { should eq 'WAITING' }
+end
+```
+##### Test that an EMR Cluster state is `RUNNING`.
+```ruby
+describe aws_emr_cluster('CLUSTER_ID') do
+  its ('state') { should eq 'RUNNING' }
+end
+```
+## Matchers
+
+This InSpec audit resource has the following special matchers. For a full list of available matchers, please visit our [matchers page](https://www.inspec.io/docs/reference/matchers/).
+
+#### exist
+
+The control will pass if the describe returns at least one result.
+
+Use `should` to test the entity should not exist.
+```ruby
+describe aws_emr_cluster('CLUSTER_ID') do
+  it { should exist }
+end
+```
+
+Use `should_not` to test the entity should not exist.
+```ruby
+describe aws_emr_cluster('cCLUSTER_ID') do
+  it { should_not exist }
+end
+```
+
+## AWS Permissions
+
+Your [Principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html#intro-structure-principal) will need the `elasticmapreduce:DescribeCluster`, `elasticmapreduce:DescribeSecurityConfiguration` & `elasticmapreduce:ListClusters` actions set to allow.
```

Comment:
```suggestion
Your [Principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html#intro-structure-principal) will need the `EMR:Client:DescribeClusterOutput`, `EMR:Client:GetManagedScalingPolicyOutput` actions set to allow.
```

Reply:
Removed the list clusters as this api is not being used in the singular resource.

Reply:
Done